### PR TITLE
[ImageGeneration] EulerAncestralDiscreteScheduler

### PR DESCRIPTION
**Pull request by likholat**
image

CVS-156803
CVS-158965

### CODE REVIEW

File: src/docs/SUPPORTED_MODELS.md [hunk 1]
```diff
@@ -206,6 +206,7 @@ The pipeline can work with other similar topologies produced by `optimum-intel`
           <li><a href="https://huggingface.co/philschmid/stable-diffusion-v1-4-endpoints"><code>philschmid/stable-diffusion-v1-4-endpoints</code></a></li>
           <li><a href="https://huggingface.co/naclbit/trinart_stable_diffusion_v2"><code>naclbit/trinart_stable_diffusion_v2</code></a></li>
           <li><a href="https://huggingface.co/Fictiverse/Stable_Diffusion_PaperCut_Model"><code>Fictiverse/Stable_Diffusion_PaperCut_Model</code></a></li>
+          <li><a href="https://huggingface.co/stabilityai/sdxl-turbo"><code>stabilityai/sdxl-turbo</code></a></li>
```
Comment (ilya-lavrenov) [inline src/docs/SUPPORTED_MODELS.md:209]: please, move to `Stable Diffusion XL` section

File: src/cpp/src/image_generation/schedulers/euler_ancestral_discrete.cpp [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    // in diffusers, it's https://github.com/huggingface/diffusers/blob/v0.31.0/src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_img2img.py#L650
+    {
+        size_t init_timestep = std::min<size_t>(num_inference_steps * strength, num_inference_steps);
+        size_t t_start = std::max<size_t>(num_inference_steps - init_timestep, 0);
+        // keep original timesteps
+        m_schedule_timesteps = m_timesteps;
+        // while return patched ones by 'strength' parameter
+        m_timesteps = std::vector<int64_t>(m_timesteps.begin() + t_start, m_timesteps.end());
+        m_begin_index = t_start;
+    }
```
[the remark this answers is not in the document]
Comment (likholat) [reply src/cpp/src/image_generation/schedulers/euler_ancestral_discrete.cpp:152]: Input image: 
image
```
prompt = "robotic racoon with wings"
strength=0.8
num_inference_steps=20
```

Result:
image

File: src/cpp/src/image_generation/schedulers/types.cpp [hunk 3]
```diff
@@ -57,6 +57,8 @@ void read_json_param(const nlohmann::json& data, const std::string& name, Schedu
             param = Scheduler::FLOW_MATCH_EULER_DISCRETE;
         else if (scheduler_type_str == "PNDMScheduler")
             param = Scheduler::PNDM;
+        else if (scheduler_type_str == "EulerAncestralDiscreteScheduler")
+            param = Scheduler::EULER_ANCESTRAL_DISCRETE;
```
[the remark this answers is not in the document]
Comment (likholat) [reply src/cpp/src/image_generation/schedulers/types.cpp:61]: Done

File: src/docs/SUPPORTED_MODELS.md (hunk 1 above)
Comment (likholat) [reply src/docs/SUPPORTED_MODELS.md]: Done

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**